wallet/src/ui/app/components/sui-apps/index.tsx

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+import { useCallback, useEffect, useState } from 'react';
+
+import SuiApp, { SuiAppEmpty } from './SuiApp';
+import Button from '_app/shared/button';
+import ExplorerLink from '_components/explorer-link';
+import { ExplorerLinkType } from '_components/explorer-link/ExplorerLinkType';
+import Icon, { SuiIcons } from '_components/icon';
+import LoadingIndicator from '_components/loading/LoadingIndicator';
+import { useAppSelector, useAppDispatch } from '_hooks';
+import { getCuratedApps } from '_redux/slices/dapps';
+import { mintDemoNFT } from '_redux/slices/sui-objects';
+
+import type { SerializedError } from '@reduxjs/toolkit';
+
+import st from './Playground.module.scss';
+
+function AppsPlayGround() {
+    const [mintInProgress, setMintInProgress] = useState(false);
+    const [mintStatus, setMintStatus] = useState<boolean | null>(null);
+    const [mintError, setMintError] = useState<string | null>(null);
+    const dispatch = useAppDispatch();
+
+    useEffect(() => {
+        dispatch(getCuratedApps()).unwrap();
+        let timeout: number;
+        if (mintStatus !== null) {
+            timeout = window.setTimeout(() => setMintStatus(null), 3000);
```

Comment:
I'm not quite sure why this timeout is 3 seconds.

Reply:
Did you follow-up with this? IMO we should just wait for network response before un-disabling the button.